src/Data/Singletons/Prelude/Functor.hs

```diff
@@ -194,13 +195,8 @@ $(singletonsOnly [d|
   void :: Functor f => f a -> f ()
   void x = () <$ x
 
-  -- deriving instance Functor ((,) a)
-  instance Functor ((,) a) where
-      fmap f (x,y) = (x, f y)
-
-  -- deriving instance Functor Down
-  instance Functor Down where
-    fmap f (Down x) = Down (f x)
+  deriving instance Functor ((,) a)
```

Comment:
Why are these here? Maybe there are module dependencies around `Down`, but the tuple instance should be in `Monad.Internal`, no?

Reply:
I suppose so. I'll try moving the `(,)` instances to `Monad.Internal` and see if anything blows up.

Reply:
Ugh, _now_ I remember why I didn't do this before. It's because these instances use `Monoid`, but `Data.Singletons.Prelude.Monad.Internal` occurs much lower in the dependency graph than `Data.Singletons.Prelude.Monoid`. It would be possible to do even more internal module reorganization to allow this, but I'm not sure it's worth it.